BookAPI/Models/Book.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace BookAPI.Models
+{
+    using System.ComponentModel.DataAnnotations;
+    public class Book
+    {
+        private static object sync = new object();
+        private static int _globalCount;
+
+        public Book(string title, string author, int numberOfPages, int year)
+        {
+            Title = title;
+            Author = author;
+            NumberOfPages = numberOfPages;
+            Year = year;
+            lock (sync)
```

Comment:
I'm surprised to see lock here. Now I'm curious. Would you be able to simulate case when it will be duplicated id`s or any other similar issue?